contracts/manifold/collectible/ERC721Collectible.sol

```diff
@@ -10,308 +10,350 @@ import "@openzeppelin/contracts/utils/Strings.sol";
 
-  /**
-   * @dev override if you want to perform different mint functionality
-   */
-  function _mint(address creatorContractAddress, uint256 instanceId, address to, uint16 amount) internal virtual {
-    CollectibleInstance storage instance = _getInstance(creatorContractAddress, instanceId);
+        return string(abi.encodePacked(instance.baseURI, Strings.toString(mintOrder)));
+    }
 
-    if (amount == 1) {
-      instance.purchaseCount++;
+    /**
+    * @dev See {IERC721CreatorExtensionApproveTransfer-setApproveTransfer}
+    */
+    function setApproveTransfer(
+      address creatorContractAddress,
+      bool enabled
+    ) external override creatorAdminRequired(creatorContractAddress) {
+        require(
+            ERC165Checker.supportsInterface(creatorContractAddress, type(IERC721CreatorCore).interfaceId),
+            "creator must implement IERC721CreatorCore"
+        );
+        IERC721CreatorCore(creatorContractAddress).setApproveTransferExtension(enabled);
+    }
 
-      // Mint token
-      uint256 tokenId = IERC721CreatorCore(creatorContractAddress).mintExtension(to);
-      _tokenIdToTokenClaimMap[creatorContractAddress][tokenId] = TokenClaim(uint224(instanceId), instance.purchaseCount);
-      emit Unveil(creatorContractAddress, instanceId, instance.purchaseCount, tokenId);
-    } else {
-      uint32 tokenStart = instance.purchaseCount + 1;
-      instance.purchaseCount += amount;
+    /**
+    * @dev See {IERC721CreatorExtensionApproveTransfer-approveTransfer}.
+    */
+    function approveTransfer(address, address from, address, uint256 tokenId) external view override returns (bool) {
+        // always allow mints
+        if (from == address(0)) {
+            return true;
+        }
+        
+        uint256 instanceId = _tokenIdToTokenClaimMap[msg.sender][tokenId].instanceId;
+        require(instanceId != 0, "Token not found");
 
-      // Mint token
-      uint256[] memory tokenIds = IERC721CreatorCore(creatorContractAddress).mintExtensionBatch(to, amount);
+        CollectibleInstance storage instance = _getInstance(msg.sender, instanceId);
 
-      for (uint32 i = 0; i < tokenIds.length; ) {
-        emit Unveil(creatorContractAddress, instanceId, tokenStart + i, tokenIds[i]);
+        return !instance.isTransferLocked;
+    }
 
-        _tokenIdToTokenClaimMap[creatorContractAddress][tokenIds[i]] = TokenClaim(uint224(instanceId), tokenStart + i);
-        unchecked {
-          i++;
+    /**
+    * @dev override if you want to perform different mint functionality
+    */
+    function _mint(address creatorContractAddress, uint256 instanceId, address to, uint16 amount) internal virtual {
+        CollectibleInstance storage instance = _getInstance(creatorContractAddress, instanceId);
+
+        if (amount == 1) {
+            uint256 tokenId;
+            if (instance.contractVersion >= 3) {
+                uint80 tokenData = uint56(instanceId) << 24 | uint24(++instance.purchaseCount);
```

Comment:
It would be nice for code clarity if this was tucked away in a separate function call or similar like -

```
tokenDataPack(uint256 instanceId, uint256 mintOrder): returns uint280 {
  return uint80(uint56(instanceId) << 24 | uint24(mintOrder))
}

tokenDataUnpack(uint80) returns(uint56,uint24) {} 
 ```

Reply:
Ya, but trying to reduce gas cost via another indirection/function call.  If you want to address this later, feel free.  Not going in this PR.